include/linker/devicetree_regions.h

```diff
@@ -7,6 +7,36 @@
  * Generate memory regions from devicetree nodes.
  */
 
+/**
+ * @brief Get the linker memory-region name
+ *
+ * This attempts to use the zephyr,linker-region property, falling back
+ * to the node path if it doesn't exist.
+ *
+ * Example devicetree fragment:
+ *
+ *     / {
+ *             soc {
+ *                     sram1: memory@2000000 {
+ *                         zephyr,linker-region = "MY_NAME";
+ *                     };
+ *                     sram2: memory@2001000 { ... };
+ *             };
+ *     };
+ *
+ * Example usage:
+ *
+ *    DT_NODE_LINKER_REGION(DT_NODELABEL(sram1)) // "MY_NAME"
+ *    DT_NODE_LINKER_REGION(DT_NODELABEL(sram2)) // "/soc/memory@2001000"
+ *
+ * @param node_id node identifier
+ * @return the name of the linker region the node will generate
+ */
+#define DT_NODE_LINKER_REGION(node_id)				     \
```

Comment:
Ugh, this isn't new in this PR, but I just noticed that this header is using the `DT_` namespace outside of devicetree.h

I'm so sorry to be a pain yet again, but I think this namespace should be reserved for that header. I've sent https://github.com/zephyrproject-rtos/zephyr/pull/37484 for the existing macro and would appreciate it if you could not add another one here. I suggested the `LINKER_` prefix there; is that OK with you?

Reply:
> this header is using the DT_ namespace outside of devicetree.h

Unresolving this one as it's still defined as DT_NODE_LINKER_REGION